src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js

```diff
@@ -526,6 +528,16 @@ function ComposerWithSuggestions({
         [blur, focus, prepareCommentAndResetComposer, replaceSelectionWithText],
     );
 
+    useEffect(() => {
+        if (ReportUtils.getCommentLength(value) <= CONST.MAX_COMMENT_LENGTH) {
+            if (hasExceededMaxCommentLength) {
+                setExceededMaxCommentLength(false);
+            }
+            return;
+        }
+        setExceededMaxCommentLength(true);
+    }, [value, setExceededMaxCommentLength, hasExceededMaxCommentLength]);
```

Comment:
I don't like this approach - I think passing setting state method down to the child component and using it there directly to change the state of the parent is an antipattern. You should use something like `onExceededMaxCommentLength` event handler and handle it in the parent instead.

Reply:
@burczu I think we do not need to rename these two props, because we also call ```setExceededMaxCommentLength(false)``` when the input `s length < 10K. So it does not make sense if we call ```onMaxLengthExceeded(false)```

Reply:
@burczu Updated based on your suggestion